src/main/java/org/opensearch/knn/index/codec/luceneonfaiss/FaissHNSW.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.luceneonfaiss;
+
+import lombok.Getter;
+import org.apache.lucene.store.IndexInput;
+
+import java.io.IOException;
+
+/**
+ * Ported implementation of the FAISS HNSW graph search algorithm.
+ * While it follows the same steps as the original FAISS implementation, differences in how the JVM and C++ handle floating-point
+ * calculations can lead to slight variations in results. However, such cases are very rare, and in most instances, the results are
+ * identical to FAISS. Even when there are ranking differences, they do not impact the precision or recall of the search.
+ * For more details, refer to the [FAISS HNSW implementation](
+ * <a href="https://github.com/facebookresearch/faiss/blob/main/faiss/impl/HNSW.h">...</a>).
+ */
+@Getter
+public class FaissHNSW {
```

Comment:
If your intention is to make this feature experiemental - please add in the following - 

```
/**
..
..
 * @opensearch.experimental
*/

@ExperimentalApi
@Getter
public class FaissHNSW {
...
```